File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {

        //the algorithm inorder traversal of binary search tree is always asccending order 

        //while doing the inorder traversal keep an counter variable so when it reaches k resturn the result 
     List<Integer> arr = new ArrayList<>();
     dfs(root,arr);
     return arr.get(k-1);
        
    }

    public void dfs(TreeNode  node ,List<Integer>arr ){
        if(node == null){
            return;
        }
        dfs(node.left , arr);
        arr.add(node.val);
        dfs(node.right,arr);


    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        q=[(root,1)]
        maxdepth=1

        while q:
            ele , level = q.pop(0)
            maxdepth = max(maxdepth,level)
            if ele.left:
                q.append((ele.left,level+1))
            if ele.right:
                q.append((ele.right , level+1))
            
            

        return maxdepth
            



       
        